File: hub-ride-frontend/src/types/address.ts
export type AddressKind = "HUB" | "POPULAR" | "CUSTOM";

export type Address = {
  id: string;
  label: string;
  fullAddress: string;
  latitude: number;
  longitude: number;
  h3Index: string;
  kind: AddressKind | string;
};

export type SelectedPlace = {
  label: string;
  fullAddress?: string;
  lat: number;
  lng: number;
  h3Index?: string;
  kind?: string;
};

File: hub-ride-backend/src/main/java/com/hubride/module/address/entity/Address.java
package com.hubride.module.address.entity;

import jakarta.persistence.*;
import lombok.*;
import java.util.UUID;

@Entity
@Table(name = "addresses")
@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Address {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @Column(name = "label", nullable = false, length = 200)
    private String label;

    @Column(name = "full_address", nullable = false, columnDefinition = "TEXT")
    private String fullAddress;

    @Column(name = "latitude", nullable = false)
    private Double latitude;

    @Column(name = "longitude", nullable = false)
    private Double longitude;

    @Column(name = "h3_index", nullable = false, length = 20)
    private String h3Index;

    @Column(name = "kind", nullable = false, length = 20)
    @Builder.Default
    private String kind = "POPULAR";
}
